addon/mixins/debounced-response.js

```diff
@@ -1,16 +1,19 @@
 import Ember from 'ember';
 
 export default Ember.Mixin.create({
-  debounce: function (handler) {
-    return () => {
-      if (!this.isScheduled) {
-        this.isScheduled = true;
+  debounce(handler) {
+    const self = this;
```

Comment:
If you know which args you need access to, you can simply destructure them and give them a name. For example:

```
return (...args) => {
  Ember.run(this, handler, ...args)
}
```